lib/interpolateName.ts

```diff
@@ -99,9 +108,17 @@ function interpolateName(loaderContext, name, options = {}) {
   }
 
   if (
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore LoaderContext doesn't even have options defined on it?
+    // If we chagned this to be `loaderContext.getOptions()` it would still not have
+    // the customInterpolateName function defined on it.
     typeof loaderContext.options === "object" &&
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     typeof loaderContext.options.customInterpolateName === "function"
   ) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     url = loaderContext.options.customInterpolateName.call(
```

Comment:
```suggestion
  interface ILoaderContextOptions {
    customInterpolateName(
      this: LoaderContext<object>,
      url: string,
      name: string | ((resourcePath: string, resourceQuery?: string) => string),
      options?: IInterpolateNameOptions
    ): string;
  }
  type LegacyLoaderContext = LoaderContext<object> & {
    options?: ILoaderContextOptions;
  };
  const loaderContextOptions: ILoaderContextOptions | undefined = (
    loaderContext as LegacyLoaderContext
  ).options;
  if (
    typeof loaderContextOptions === "object" &&
    typeof loaderContextOptions.customInterpolateName === "function"
  ) {
    url = loaderContextOptions.customInterpolateName.call(
```

Reply:
`LegacyLoaderContext` may not be an appropriate name. Is this actually a legacy thing? Maybe `InternalLoaderContext`.

Reply:
loaderContext here comes from `webpack` however loaderContext in webpack 5 does not even have the properties `customInterpolateName` or let alone the `options` property. To get the loader in question's options, you need to use `loaderContext.getOptions()` (webpack 5 change), and even if you did, the aformentioned properties still do not live in `loaderContext.getOptions`.

Reply:
@alexander-akait @sokra Are these legacy API's on `loaderContext` here? I'm confused where `customInterpolateName` even comes from.